### PR TITLE
Amrfm 874 Detect obstacles with kinect and stop

### PR DESCRIPTION
**Pull request by rdeniza**
We're bringing the feasibility check back. So now, with using kinect, we can detect obstacles under the lidar field and stop before hitting them. This is by default turned off, one can turn it on by adding
feasibility_check: True
to the corresponding FollowPath config.

### CODE REVIEW

File: teb_local_planner/include/teb_local_planner/teb_config.h [hunk 1]
```diff
@@ -266,6 +267,7 @@ class TebConfig
     trajectory.force_reinit_new_goal_dist = 1;
     trajectory.force_reinit_new_goal_angular = 0.5 * M_PI;
     trajectory.feasibility_check_no_poses = 5;
+    trajectory.feasibility_check = false;
```
Comment (jplapp) [inline teb_local_planner/include/teb_local_planner/teb_config.h:270]: it seems like above it is specified as an integer?

File: teb_local_planner/src/optimal_planner.cpp [hunk 2]
```diff
@@ -1281,4 +1281,18 @@ bool TebOptimalPlanner::isTrajectoryFeasible(dwb_critics::ObstacleFootprintCriti
   return true;
 }
 
+bool TebOptimalPlanner::isPoseValid(geometry_msgs::msg::Pose2D pose2d, dwb_critics::ObstacleFootprintCritic* costmap_model,
```
Comment (jplapp) [inline teb_local_planner/src/optimal_planner.cpp:1284]: where is this function used?

**Review by jplapp — CHANGES_REQUESTED** [review]
The changes here look good, but I'm not sure if everything is included, please check

File: teb_local_planner/include/teb_local_planner/teb_config.h (hunk 1 above)
Comment (rdeniza) [reply teb_local_planner/include/teb_local_planner/teb_config.h:270]: Ah yes, fixed

File: teb_local_planner/src/optimal_planner.cpp (hunk 2 above)
Comment (rdeniza) [reply teb_local_planner/src/optimal_planner.cpp:1284]: It seems usage disappeared when I merge foxy-devel into it. I've fixed it

File: teb_local_planner/include/teb_local_planner/teb_config.h (hunk 1 above)
Comment (jplapp) [reply teb_local_planner/include/teb_local_planner/teb_config.h:270]: is that the way to fix it? In above file, the setting is defined as 
grp_trajectory.add("feasibility_check",   int_t,   0,
  "Specify up to which pose on the predicted plan the feasibility should be checked each sampling interval",
  20, 0, 50)

File: teb_local_planner/src/optimal_planner.cpp [hunk 3]
```diff
@@ -106,6 +106,7 @@ void TebOptimalPlanner::initialize(nav2_util::LifecycleNode::SharedPtr node, con
   vel_goal_.second.linear.y = 0;
   vel_goal_.second.angular.z = 0;
   initialized_ = true;
+  setVisualization(visual);
```
Comment (jplapp) [inline teb_local_planner/src/optimal_planner.cpp:109]: is this related?

**Review by jplapp — CHANGES_REQUESTED** [review]
nearly there, please re-check comments about type

File: teb_local_planner/src/optimal_planner.cpp (hunk 3 above)
Comment (andriimaistruk) [reply teb_local_planner/src/optimal_planner.cpp:109]: It was in the commit cherry-picked from upstream

**Review by jplapp — APPROVED** [review]

**Comment by jplapp** [comment]
thanks @andriimaistruk  for finishing this
